gradio/components/upload_button.py

```diff
@@ -115,14 +124,31 @@ def example_inputs(self) -> Any:
                 "https://github.com/gradio-app/gradio/raw/main/test/test_files/sample_file.pdf"
             ]
 
+    def _process_single_file(self, f: dict[str, Any]) -> bytes | str:
+        file_name, data, is_file = (
+            f["name"],
+            f["data"],
+            f.get("is_file", False),
+        )
+        if self.type == "filepath":
+            file = tempfile.NamedTemporaryFile(delete=False, dir=self.GRADIO_CACHE)
```

Comment:
Actually this isn't really backwards compatible because you can't really access the file anymore. Like what if a demo is using `file.read()` or something? Maybe we should just get rid of the file concept altogether and return the string?